### ISSUE TITLE
marathon doesn't start correctly post-reboot on single control-node installs

### ISSUE THREAD

**Issue by ryane** [CiscoCloud/microservices-infrastructure#533]
After running `playbooks/reboot-hosts.yml', marathon may not start correctly. Frequently, it ends up stuck in a loop where it is repeatedly trying to connect to zookeeper:

```
Jun 26 20:48:21 default marathon[765]: [2015-06-26 20:48:20,859] INFO Connecting to Zookeeper... (mesosphere.marathon.Main$:37)
Jun 26 20:48:21 default marathon[765]: [2015-06-26 20:48:20,859] INFO Initiating client connection, connectString=zookeeper.service.consul:2181 sessionTimeout=10000 watcher=com.twitter.common.zookeeper.ZooKeeperClient$3@cd73642 (org.apache.zookeeper.ZooKeeper:379)
Jun 26 20:48:21 default marathon[765]: [2015-06-26 20:48:20,877] WARN Unable to connect to Zookeeper, retrying... (mesosphere.marathon.Main$:43)
```

But, zookeeper is up and running at the time: 

```
echo ruok | nc zookeeper.service.consul 2181
imok
```

Simply restarting marathon gets things going again. This *seems* to only affect installs with a single control node (a default vagrant build, for example).

**Comment by stevendborrelli** [comment]
Is this a bug we should open with Marathon?

**Comment by ryane** [comment]
When consul is in maintenance mode, `zookeeper.service.consul` cannot be resolved. Marathon starts before the `reboot-hosts` playbook disables maintenance mode post-reboot. And it appears Marathon cannot recover from this once it has started - even when the host can subsequently resolve  `zookeeper.service.consul`. So, yeah, I think it would be nice to have it fixed in marathon - I'll open an issue. Should we put in a workaround for now? Trigger a marathon restart after disabling consul maintenance mode in the `reboot-hosts`playbook? Or change the marathon service definition to use a wrapper script that ensures it can reach zookeeper before starting marathon? Something else?

**Comment by stevendborrelli** [comment]
Workaround sounds good. It takes a while for Marathon to fix issues, and version 0.9 will refactor the storage backend, so we don't know how it will affect ZK connectivity.
